web/src/state_data.ts

```diff
@@ -21,29 +21,49 @@ export const narrow_term_schema = z.object({
 export type NarrowTerm = z.output<typeof narrow_term_schema>;
 // Sync this with zerver.lib.events.do_events_register.
 
+const position_schema = z.enum([
+    "top",
+    "left",
+    "right",
+    "bottom",
+    "left_bottom",
+    "viewport_center",
+]);
+
+export type Position = z.infer<typeof position_schema>;
```

Comment:
This gets stored in variables named `placement`, so let’s reuse the existing name instead of inventing a synonym: `placement_schema` and `Placement`.